flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/AbstractHaServices.java

```diff
@@ -242,28 +251,15 @@ public CompletableFuture<Void> globalCleanupAsync(JobID jobID, Executor executor
     }
 
     private LeaderElection createLeaderElection(String leaderName) throws Exception {
-        final DefaultLeaderElectionService leaderElectionService =
-                new DefaultLeaderElectionService(createLeaderElectionDriverFactory(leaderName));
-        leaderElectionService.startLeaderElectionBackend();
-
-        closeableRegistry.registerCloseable(leaderElectionService);
-        // the leaderName which is passed as a contenderID here is not actively used within the
-        // DefaultLeaderElectionService for now - this will change in a future step where the
-        // DefaultLeaderElectionService will start to use MultipleComponentLeaderElectionDriver
-        // instead of LeaderElectionDriver and fully replace
-        // DefaultMultipleComponentLeaderElectionService (FLINK-31783)
-        return leaderElectionService.createLeaderElection("unused-contender-id");
-    }
+        synchronized (lock) {
+            if (leaderElectionService == null) {
```

Comment:
We have to start the leader election lazily here. Otherwise, `JobManagerHAProcessFailureRecoveryITCase` would stall. It looks like the reason is that we create `TaskManagers` sequentially there which rely on the `ResourceManager` being around (for setting up the Rpc bind address).

Creating the driver in the `AbstractHaServices` constructor will trigger the leadership grant earlier without the `ResourceManager` being instantiated, yet. This makes the address resolution fail and the test won't succeed.

Therefore, I kept it like that for now (because that's how the `DefaultMultipleComponentLeaderElectionService` was instantiated as well). But I still have to investigate whether it's actually true and whether we can see the same behavior in `release-1.17`.